### DISCUSSION TITLE
Hash consistency

### DISCUSSION THREAD

**Discussion by troolee** [laurentpayot/minidenticons#5] [Ideas]
Hey, I really like the lib just wondering if there is any idea to keep hashing algorithm consistent?

The problem that I experience is that pretty much after each update icons change for the same id. Alternatively maybe it could be possible to specify custom hashing function that would return an input for the render function.

**Comment by laurentpayot** [comment]
Hi @troolee. Sorry for the recurring hashing algorithm changes. I always make major version bumps when these changed occurr to prevent automatic updates by npm clients.

The recent hashing algorithm change was the last ever (I think! :crossed_fingers:). I replaced long and uncompressible FNV-1a magic numbers by `5` in the hash function to reduce its size. I cannot see how simpler a _decent_ hash function could be.
But I also changed how the patterns are drawn: instead of calculating modulos it nows simply checks low weight bits of the hash. So having kept the same hash algorithm this time would not have prevented identicons to change. Hopefully this drawing method change was the last ever too.
I also used this major version bump to change the names from `identicon` to `minidenticon` everywhere.

I was wondering if an extra parameter for a custom hash function was useful or not. As you requested it, I just released v4.2.0 with that added feature :package:  

I think that after reaching v4.2.0 Minidenticons is now at a very stable state. I will step back a little to ensure that :wink:  

Cheers,
Laurent

**Reply by troolee** [reply]
Thank you 🙏
